src/Tizen.Security.PrivacyPrivilegeManager/Tizen.Security/PrivacyPrivilegeManager.cs

```diff
@@ -203,23 +386,27 @@ internal ResponseContext(string privilege)
                 add
                 {
                     _ResponseFetched += value;
+                    if (!s_responseMap.ContainsKey(_privilege))
+                    {
+                        s_responseMap[_privilege] = new ResponseContext(_privilege);
+                    }
                 }
 
                 remove
                 {
                     _ResponseFetched -= value;
-                    if (_ResponseFetched == null)
-                    {
-                        s_responseMap.Remove(_privilege);
-                    }
                 }
             }
 
             private event EventHandler<RequestResponseEventArgs> _ResponseFetched;
 
             internal void FireEvent(CallCause _cause, RequestResult _result)
             {
-                _ResponseFetched?.Invoke(null, new RequestResponseEventArgs() { cause = _cause, result = _result, privilege = _privilege });
+                _ResponseFetched?.Invoke(null, new RequestResponseEventArgs { cause = _cause, result = _result, privilege = _privilege });
+            }
+            ~ResponseContext()
+            {
+                s_responseMap.Remove(_privilege);
```

Comment:
It has race condition issue, destructor was executed on GC Thread.

and it seem to be dead code, 
because, this object was strongly referenced by s_responseMaps

If _privilege was not existed in s_responseMaps, you do not need to remove from s_responseMap
If _privilege was existed in s_responseMaps, This object never destroyed

Reply:
done

Reply:
??